mproton_darwin.m

```diff
@@ -0,0 +1,255 @@
+    [mainMenu addItem:appMenuItem];
+    
+    return mainMenu;
+}
+
+static NSStatusItem * createMenuExtra() {
+    NSStatusItem * statusItem = [[NSStatusBar systemStatusBar] statusItemWithLength:NSVariableStatusItemLength];
+    statusItem.button.title= @"🌯";
+     
+    NSMenu *menu = [[NSMenu alloc] init];
+    [menu addItem:[NSMenuItem separatorItem]]; // A thin grey line
+    [menu addItemWithTitle:@"Quit" action:@selector(terminate:) keyEquivalent:@"Q"];
+    statusItem.menu = menu;
+    
+    return statusItem;
+}
+
+static WKWebView * createWebView(NSRect frame, id handler) {
+    WKWebViewConfiguration *theConfiguration = [[WKWebViewConfiguration alloc] init];
+    [theConfiguration.preferences setValue:@YES forKey:@"developerExtrasEnabled"];
+    [theConfiguration.preferences setValue:@YES forKey:@"fullScreenEnabled"];
+    [theConfiguration.preferences setValue:@YES forKey:@"javaScriptCanAccessClipboard"];
+    [theConfiguration.preferences setValue:@YES forKey:@"DOMPasteAllowed"];
+    
+    WKUserContentController *userContentController = [[WKUserContentController alloc] init];
+    theConfiguration.userContentController = userContentController;
+    
+    WKWebView *webView = [[WKWebView alloc]
+                          initWithFrame:frame
+                          configuration:theConfiguration];
+
+    // NSURL *nsurl=[NSURL URLWithString:@"http://www.apple.com"];
+    // NSURLRequest *nsrequest=[NSURLRequest requestWithURL:nsurl];
+    // [webView loadRequest:nsrequest];
+    
+    // NSURL *url = [NSURL URLWithString:@"file:///Users/iokulist/Github/okigan/proton2/protonappui/dist/index.html"];
+    // NSString *html = [NSString stringWithContentsOfURL:url encoding:NSUTF8StringEncoding error:nil];
+    // NSURL *baseUrl = [NSURL URLWithString:@""];
+    
+    // [webView loadHTMLString:html baseURL:baseUrl];
+        
+    return webView;
+}
+
+- (void) applicationWillFinishLaunching: (NSNotification *)notification {
+    NSLog(@"in applicationWillFinishLaunching");
+    id appName = [[NSProcessInfo processInfo] processName];
+    
+    self.window = g_appContext.window;
+    
+    self.mainMenu = createMainMenu(appName);
+    [NSApp setMainMenu:self.mainMenu];
+    
+    self.statusItem = g_appContext.statusItem;
+    
+    self.window.contentView = g_appContext.webView;
+    
+    [NSApp setActivationPolicy:NSApplicationActivationPolicyRegular];
+    [NSApp activateIgnoringOtherApps:YES];
+}
+
+- (void)applicationDidFinishLaunching:(NSNotification *)notification {
+    NSLog(@"in applicationDidFinishLaunching");
+    
+    [NSApp setActivationPolicy:NSApplicationActivationPolicyRegular];
+    [NSApp activateIgnoringOtherApps:YES];
+}
+
+- (void)  MenuExtraCallback: (NSMenuItem*) sender {
+    const char *extractedExpr = [@"p111" UTF8String];
+    const char *extractedExpr2 = [@"p222" UTF8String];
+    struct goTrampoline_return result = goTrampoline(17, (char*)(extractedExpr), (char*)(extractedExpr2));
```

Comment:
Probably use a separate trampoline for the menu actions. The arguments are different for the functions.

Reply:
Umm, still thinking about this one